dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/CompanyDataControllerGetCompaniesEndpointTest.kt

```diff
@@ -0,0 +1,287 @@
+                onlyCompanyNames = false,
+            ).isEmpty(),
+            "The posted company was found in the query results.",
+        )
+        apiAccessor.companyDataControllerApi.existsIdentifier(
+            CompanyDataControllerApi.IdentifierType_existsIdentifier.permId,
+            uploadInfo.inputCompanyInformation.identifiers.first().identifierValue,
+        )
+        uploadTestDataSet(uploadInfo.actualStoredCompany.companyId)
+        assertTrue(
+            apiAccessor.companyDataControllerApi.getCompanies(
+                apiAccessor.frameworkData,
+                searchString = uploadInfo.inputCompanyInformation.identifiers.first().identifierValue,
+                onlyCompanyNames = false,
+            ).any { it.companyId == uploadInfo.actualStoredCompany.companyId },
+            "The posted company could not be found in the query results when querying for its first identifiers value.",
+        )
+    }
+
+    private fun testThatSearchForCompanyIdentifierWorks(identifier: CompanyIdentifier) {
+        val searchResponse = apiAccessor.companyDataControllerApi.getCompanies(
+            dataTypes = apiAccessor.frameworkData,
+            searchString = identifier.identifierValue,
+            onlyCompanyNames = false,
+
+        )
+            .toMutableList()
+        // The response list is filtered to exclude results that match in account of another identifier having
+        // the required value but the looked for identifier type does not exist (This happens due to the test
+        // data having non-unique identifier values for different identifier types)
+        searchResponse.retainAll {
+            it.companyInformation.identifiers.any {
+                    identifierInResponse ->
+                identifierInResponse.identifierType == identifier.identifierType
+            }
+        }
+        assertTrue(
+            searchResponse.all { results -> results.companyInformation.identifiers.any { it == identifier } },
+            "The search by identifier returns at least one company that does not contain the looked" +
+                "for value $identifier.",
+        )
+    }
+
+    @Test
+    fun `search for all identifier values and check if all results contain the looked for value`() {
+        apiAccessor.jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.Uploader)
+        val testString = UUID.randomUUID().toString()
+        val testCompanyList = listOf(
+            CompanyInformation(
+                company1, "",
+                listOf(
+                    apiAccessor.createCompanyIdentifier(CompanyIdentifier.IdentifierType.isin, "Isin$testString"),
+                    apiAccessor.createCompanyIdentifier(CompanyIdentifier.IdentifierType.lei, "Lei$testString"),
+                ),
+                "",
+                listOf(company1 + testString),
+            ),
+        )
+        for (company in testCompanyList) {
```

Comment:
for-loop makes no sense given that there is only one company in the `testCompanyList`